src/testing.c

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "testing.h"
+
+#include "console.h"
+
+// A flag used to record the return value of the most recent test
+static int recent_test_result = 1;
+
+// A function to time a specific test
+// Returns the amount of time that elapsed during the test
+clock_t time_test(tester test) {
+	clock_t start = clock_read();
+	recent_test_result = (test)();
+	clock_t end = clock_read();
+	return clock_diff(start, end);
+}
+
+// A test runner
+// Returns true if a test was executed in a passing amount of time
+int run_test(struct test_runner *runner) {
```

Comment:
should probably add a flag here and to run_all_tests that allows results of tests to be printed out (so can run all tests and just show if they passed or not)